cli/src/main/java/org/jboss/as/cli/impl/ValueTypeCompleter.java

```diff
@@ -375,15 +379,24 @@ public int complete(CommandContext ctx, String buffer, int cursor, List<String>
         try {
             handler = parse(buffer);
         } catch (CommandFormatException e) {
-            // TODO add logging here
+            if (LOG.isEnabled(Level.WARN)) {
+                LOG.log(Level.WARN, e.getLocalizedMessage(), e);
+            }
             return -1;
         }
-        final Collection<String> foundCandidates = handler.getCandidates(propDescr);
-        if(foundCandidates.isEmpty()) {
+        try {
+            final Collection<String> foundCandidates = handler.getCandidates(propDescr);
+            if (foundCandidates.isEmpty()) {
+                return -1;
+            }
+            candidates.addAll(foundCandidates);
+            return handler.getCompletionIndex();
+        } catch (Throwable th) {
```

Comment:
Should this be RuntimeException? You can't really recover from Error and if someone changes something such that the try block throws a non-Runtime Exception then they probably should have to think about whether this catch logic is what they want.

Not a big deal, it's just that I've fixed (or introduced myself!!) a fair number of bugs that pop up later due to old code using catch (Exception) or catch (Throwable).